src/Tools/NodeDocumentationMarkdownGenerator/MarkdownHandler.cs

```diff
@@ -169,37 +170,62 @@ private static string GetDefaultContent(string nodeName)
 
         private static bool TrySaveImage(
             DirectoryInfo imgDir, string imgName, 
-            ImageOptimizer optimizer, FileInfo fileInfo, 
+            ImageOptimizer optimizer,bool compressGifs, FileInfo fileInfo, 
             out string outputImagePath)
         {
             outputImagePath = string.Empty;
             var imageFileInfo = imgDir.GetFiles($"{imgName}.*").FirstOrDefault();
 
             if (!imageFileInfo.Exists) return false;
 
+            
+
             try
             {
                 using (Image image = Image.FromFile(imageFileInfo.FullName))
                 using (MemoryStream m = new MemoryStream())
                 {
                     image.Save(m, image.RawFormat);
-
-                    if (optimizer != null)
+                    var oldSize = m.Length;
+                    var ext = imageFileInfo.Extension.ToLower();
+                    //the simple optimizer fails with animated gifs, or gifs that have already been semi optimized (different size frames)
+                    if (optimizer != null && ext != ".gif" )
                     {
                         m.Position = 0;
                         optimizer.Compress(m);
                     }
+                    if (compressGifs && ext == ".gif")
+                    {
+                        using (MagickImageCollection images = new MagickImageCollection(imageFileInfo))
+                        {
+                            //I've disabled these optimization steps for now because they are extremely slow.
+                            //quantization is much faster and produces mostly very good results. (though not lossless)
+
+                            //images.Coalesce();
+                            //images.Optimize();
+                            //images.OptimizeTransparency();
+                            //reduce color bit depth from 8 to 5.
+                            images.Quantize(new QuantizeSettings() { Colors = 32768 });
```

Comment:
Why `32768` ?
any special meaning ?

Reply:
15-bit high color ?

Reply:
yes (2^5) bits per red, green, and blue 32*32*32 = 32k colors.